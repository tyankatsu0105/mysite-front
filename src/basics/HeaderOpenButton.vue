<template>
	<button
		class="HeaderOpenButton"
		type="button"
		@click="openHeader"
	>
		Menu
	</button>
</template>

<script>
export default {
  props: {
    isOpenHeader: {
      type: Boolean
    }
  },
  methods: {
    openHeader() {
      this.$emit("open-header", true);
    }
  }
};
</script>

<style scoped lang="scss">
.HeaderOpenButton {
  position: relative;
  padding-bottom: 8px;
  font-size: 1.2rem;
  &::before {
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 2px;
    content: "";
    background-color: $color-accent;
    transition: $tst-default;
  }

  &:hover {
    &::before {
      width: calc(100% + 20px);
    }
  }
}
</style>
