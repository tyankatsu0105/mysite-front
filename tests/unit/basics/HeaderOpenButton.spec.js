import { shallowMount } from "@vue/test-utils";
import HeaderOpenButton from "@/basics/HeaderOpenButton.vue";

describe("HeaderOpenButton", () => {
  it("render", () => {
    const wrapper = shallowMount(HeaderOpenButton);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
