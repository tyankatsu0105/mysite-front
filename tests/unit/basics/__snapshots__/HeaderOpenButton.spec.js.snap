// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderOpenButton render 1`] = `
<button type="button" class="HeaderOpenButton">
  Menu
</button>
`;
